io/src/main/scala/fs2/io/tcp/Socket.scala

```diff
@@ -236,28 +236,33 @@ protected[tcp] object Socket {
         // buffer is also reset to be ready to be written into.
         def getBufferOf(sz: Int): F[ByteBuffer] =
           bufferRef.get.flatMap { buff =>
-            if (buff.capacity() < sz)
+            if (buff.capacity() < sz) {
               bufferRef
                 .modify { _ =>
                   ByteBuffer.allocate(sz)
                 }
-                .map { _.now } else {
+                .map { _.now }
+            } else {
               buff.clear()
+              buff.limit(sz)
```

Comment:
Is there any reason not to have

F.delay {
  buff.clear()
  buff.limit(sz)
  buff
}
instead

Reply:
No thanks, will fix